diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8b5ee88d..25e47bf4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,7 +13,7 @@ IF (CMAKE_COMPILER_IS_GNUCXX )
    SET(CMAKE_CXX_FLAGS_DEBUG  "-O0 -g")
    SET(CMAKE_CXX_FLAGS_RELEASE "-O3 -DNDEBUG ")

-  ADD_DEFINITIONS("-Wall -Werror -Wno-unused-variable
+  ADD_DEFINITIONS("-Wall -Wno-unused-variable
                    -Wno-unused-but-set-variable -Wno-unknown-pragmas ")
 ENDIF()

diff --git a/sophus/so2.cpp b/sophus/so2.cpp
index 44fc4a1a..2f6b5003 100644
--- a/sophus/so2.cpp
+++ b/sophus/so2.cpp
@@ -29,8 +29,7 @@ namespace Sophus

 SO2::SO2()
 {
-  unit_complex_.real() = 1.;
-  unit_complex_.imag() = 0.;
+  unit_complex_ = std::complex<double>(1., 0.);
 }

 SO2
